### PR TITLE
UAVCAN updates and fixes

### PR DESCRIPTION
**Pull request by haitomatic**
- Fix UAVCAN task return which destroys CAN socket issue (fix pulled from upstream latest commit https://github.com/PX4/PX4-Autopilot/commit/33e39d68f77222978637fddd93c3a455a1997e78)
- Add some module parameters to enable/disable UAVCAN PUBs/SUBs, UAVCAN features. 
- Allow resending UAVCAN node parameter request

### CODE REVIEW

File: src/drivers/uavcan/CMakeLists.txt [hunk 1]
```diff
@@ -40,7 +40,10 @@ set(UAVCAN_USE_CPP03 ON CACHE BOOL "uavcan cpp03")
 set(UAVCAN_PLATFORM "generic")
 
 if(CONFIG_ARCH_CHIP)
-	if(${CONFIG_ARCH_CHIP} MATCHES "kinetis")
```
Comment (pussuw) [inline src/drivers/uavcan/CMakeLists.txt:43]: Forklifting (via copy&pasting code) stuff from upstream instead of taking the full commit is not a good idea as it will cause a conflict when trying to push our changes back into upstream. 

If we need this code, simply cherry-pick the commit and make a pull request to merge (rebase) it into our fork

File: src/drivers/uavcan/uavcan_drivers/kinetis/driver/include/uavcan_kinetis/can.hpp [hunk 2]
```diff
@@ -360,18 +360,13 @@ class CanInitHelper
 	 * This function can either initialize the driver at a fixed bit rate, or it can perform
 	 * automatic bit rate detection. For theory please refer to the CiA application note #801.
 	 *
-	 * @param delay_callable    A callable entity that suspends execution for strictly more than one second.
```
Comment (pussuw) [inline src/drivers/uavcan/uavcan_drivers/kinetis/driver/include/uavcan_kinetis/can.hpp:363]: Same here. The two commits in this PR, cherry pick them fully and make a separate PR of that. When it is merged, then make another PR of the 3 remaining commits.

**Review by jlaitine — COMMENTED** [review]
Please check this PR, and close if not relevant any more, thanks 🙂
